walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,188 @@
 
 package com.google.sps;
 
+import com.google.sps.TimeRange;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  public static final int WHOLE_DAY = 24 * 60;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> meetings = new ArrayList<>();
+    Collection<String> requestedAttendees = request.getAttendees();
+    Collection<String> requestedOptionalAttendees = 
+      request.getOptionalAttendees();
+    long durationOfRequest = request.getDuration();
+
+    // Case 0: if duration too long (longer than a day), return no options
```

Comment:
what is durationOfRequest is <= 0?

Reply:
I'm confused, can you elaborate?

Reply:
Oh sorry, I mistyped a word.  You tested for the case if the duration is longer than a day. What if the a duration of -1234 gets passed in?